single_kernel_mongo/status.py

```diff
@@ -39,17 +68,93 @@ def to_waiting(self, message: str):
         """Sets status to waiting."""
         self.set_and_share_status(WaitingStatus(message))
 
-    def to_active(self, message: str | None):
-        """Sets status to active."""
-        if message is None:
-            self.set_and_share_status(ActiveStatus())
-            return
-        self.set_and_share_status(ActiveStatus(message))
-
     def to_maintenance(self, message: str):
         """Sets status to maintenance."""
         self.set_and_share_status(MaintenanceStatus(message))
 
     def to_error(self, message: str):
         """Sets status to error."""
         self.set_and_share_status(ErrorStatus(message))
+
+    def get_statuses(self) -> StatusesDict:
+        """Collects the statuses of all managers."""
+        if self.operator.name == CharmKind.MONGOD:
+            return StatusesDict(
+                {
+                    "mongodb": self.operator.mongo_manager.get_status(),
+                    "shard": self.operator.shard_manager.get_status(),
+                    "config-server": self.operator.config_server_manager.get_status(),
+                    "PBM": self.operator.backup_manager.get_status(),
+                }
+            )
+        return StatusesDict(
+            {
+                "mongodb": WaitingStatus("waiting for mongos to start")
+                if not self.operator.workload.active
+                else ActiveStatus(),
```

Comment:
I believe this doesn't include cases like:
                WaitingStatus("Waiting for secrets from config-server")
            return BlockedStatus("mongos requires TLS to be enabled.")
            return BlockedStatus("mongos has TLS enabled, but config-server does not.")
            return BlockedStatus("mongos CA and Config-Server CA don't match.")

In the original verision of the mongos charm there are various states of the mongos status [seen in update_status](https://github.com/canonical/mongos-operator/blob/9f81c03c9f23faf1975125bdf77c96ea08f916df/src/charm.py#L131-L152), it might be good to include those here?

Reply:
You're right.
I'll add this tomorrow morning.